### ISSUE TITLE
Makine Öğrenmesi Derinlikleri

### ISSUE THREAD

**Issue by MelisaYasak** [tefiq/.github#5]
1. Basit Regresyon Modelleri (Sürekli değer tahmini için) 
- [x] '': https://www.w3schools.com/python/python_ml_linear_regression.asp
- [x] Doğrusal Regresyon (Linear Regression) Teoriği (https://www.youtube.com/watch?v=Z2Gr9F3S6yU)
- [x] '': https://medium.com/machine-learning-t%C3%BCrkiye/lineer-regresyon-c7f9fb611605
- [x] '' Python ile: https://miuul.com/blog/python-ile-dogrusal-regresyon-uygulamasi
- [x] '' '': https://cerden.medium.com/python-ile-basit-regresyon-uygulamas%C4%B1-669f7b32d72c
- [x] '' '': https://www.veribilimiokulu.com/basit-regresyon-python-ile-uygulama/
- [x] Ev fiyatı tahmini (sklearn ile) Pratiği (https://www.kaggle.com/competitions/house-prices-advanced-regression-techniques)
- [x] Cross Validation: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_score.html
- [x] f_Regresion: https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.f_regression.html

2. Basit Sınıflandırma Modelleri (Etiket tahmini için)
- [ ] Lojistik Regresyon (Logistic Regression) Teoriği
- [ ] Kanser hücresi iyi/kötü sınıflandırması (Breast Cancer Dataset) ile Pratiği

3. Karar Ağaçları (if-else gibi çalışan sezgisel modeller)
- [x] Decision Trees Teoriği
- [ ] Oyun oynama kararı ("hava yağmurlu mu?" örneği) ile Pratiği (https://www.kaggle.com/competitions/hw-3-weather-forecast)
**Notlar**: Scaler kullanılacak. Diğer ağaç modelleri de kullanılacak.

4. K-En Yakın Komşu (KNN) (çok sezgisel, hesaplama ağırlıklı)
- [ ] K-Nearest Neighbors Teoriği
- [ ] El yazısı rakam tanıma (MNIST küçük subseti) ile Pratiği

5. Destek Vektör Makineleri (SVM) (ayrım sınırları daha net olsun diye)
- [ ] Support Vector Machines Teoriği
- [ ] İki sınıflı veri (örneğin çiçek türü ayırımı, Iris Dataset) ile Pratiği

6. Topluluk Yöntemleri (Ensemble Methods)
- [ ] Random Forest Teoriği
- [ ] Gradient Boosting (XGBoost gibi) Teoriği
- [ ] Titanic hayatta kalma tahmini (Kaggle’ın Titanic dataseti) ile iki modelin de pratiği

7. Temel Yapay Sinir Ağları (Neural Networks)
- [ ] Tek katmanlı ve çok katmanlı (MLP) teoriği
- [ ] Basitleştirilmiş MNIST (0-9 rakamlarını sınıflandır) ile pratiği
- [ ] CNN, RNN gibi derin öğrenme mimarileri teoriği
- [ ] CNN: basit bir imaj sınıflandırması (örneğin kedi-köpek ayrımı) ile pratiği

🎯 Genel Yöntem
Her modelde:

- Problem tanımı (Neyi öğreniyoruz?)
- Modelin iç mantığı (Nasıl çalışıyor?)
- Örnek veri seti (Nereden alıyoruz?)
- Basit kod (3-10 satırlık "fit-predict" kodu)
- Sonuç analizi (accuracy, mse gibi metrikler)

**Comment by buglss** [comment]
https://www.w3schools.com/python/python_ml_getting_started.asp

**Comment by buglss** [comment]
https://www.w3schools.com/python/python_ml_polynomial_regression.asp

**Comment by buglss** [comment]
Lineer regresyon ile ev fiyat tahmini uygulaması tamamlandı. 

https://github.com/tefiq/.github/tree/master/house-prices

**Comment by buglss** [comment]
https://www.w3schools.com/python/python_ml_decision_tree.asp tamamlandı

**Comment by MelisaYasak** [comment]
https://www.kaggle.com/code/faressayah/decision-trees-random-forest-for-beginners
